front-end/src/Views/ProjectPage.js

```diff
@@ -320,6 +326,101 @@ export default class HomePage extends Component {
+        .then(res => {
+          axios.patch(`${config.API_URL}/api/projects/${this.state.projectId}/`, { headers: { 'Content-Type': 'Application/json', 'Authorization': `Token ${getAccessToken()}` } })
+            .then(res => {
+              this.getProject(this.state.projectId);
+            })
+        }); 
+  }
+
+  acceptColabRequest = (myId) => {
+    axios.post(`${config.API_URL}/api/collaboration_requests/${myId}/accept_collaboration/`, { id: myId }, { headers: { 'Content-Type': 'Application/json', 'Authorization': `Token ${getAccessToken()}` } })
+        .then(res => {
+          axios.patch(`${config.API_URL}/api/projects/${this.state.projectId}/`, { headers: { 'Content-Type': 'Application/json', 'Authorization': `Token ${getAccessToken()}` } })
+            .then(res => {
+              this.deleteColabRequest(myId);
+            })
+        }); 
+  };
+
+  rejectColabRequest = (myId) => {
+    axios.post(`${config.API_URL}/api/collaboration_requests/${myId}/reject_collaboration/`, { id: myId }, { headers: { 'Content-Type': 'Application/json', 'Authorization': `Token ${getAccessToken()}` } })
+        .then(res => {
+          axios.patch(`${config.API_URL}/api/projects/${this.state.projectId}/`, { headers: { 'Content-Type': 'Application/json', 'Authorization': `Token ${getAccessToken()}` } })
+            .then(res => {
+              this.deleteColabRequest(myId);
+            })
+        }); 
+  };
+
+  renderColabRequests = () => {
+    const { allColabRequests } = this.state;
+    
+    var ids = [];
+    for(var i=0; i<allColabRequests.length; i++){
+      var idPair = [allColabRequests[i].id, allColabRequests[i].from_user];
+      ids.push(idPair);
+    }
+
+    // const len = uniqueIds.length;
+    // for(var k=0; k<len; k++){
+    //   axios.get(`${config.API_URL}/api/users/${uniqueIds[k]}/`, { headers: { 'Content-Type': 'Application/json', 'Authorization': `Token ${getAccessToken()}` } })
+    //   .then(res => {
+    //     let name = res.data.profile[0].name;
+    //     let mname = res.data.profile[0].middle_name;
+    //     let lastname = res.data.profile[0].last_name;
+    //     let username = name + " " + mname + " " + lastname;
+        
+    //     var joined = this.state.usernames.concat(username);
+    //     this.setState({ usernames: joined })
+    //   });
+    // }
+  
+    if (ids.length === 0) return (  <Typography variant='h6' color="textPrimary">No Collaborators</Typography>)
+    else return ids.map((item) => {
+      return (<>
+        <Typography variant="h6" color="primary" style={{ width: "100%", textAlign: "left" }}>{item[0]}</Typography>
+        <Button variant="contained" color="primary" style={{ marginLeft:"10px" }} onClick={() => this.acceptColabRequest(item[1])} > Accept </Button>
+        <Button variant="contained" color="primary" style={{ marginLeft:"10px" }} onClick={() => this.rejectColabRequest(item[1])} > Reject </Button>

```

Comment:
accept and reject collaboration requests' parameters should be item[0]. Otherwise,we send a request to accept or reject wrong collaboration  request

Reply:
This is the source of authorization problem

Reply:
Solved the issue, thanks a lot.